### PR TITLE
(feat):added hardhat with workspaces

### PR DESCRIPTION
**Pull request by 3scava1i3r**
added basic hardhat support with workspaces

To Do

- Add smart contract verification
- add post deployment scripts
- Create an "interface". In general, it has the same functionality as etherscan(write/read)

### CODE REVIEW

File: hardhat/scripts/make.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// We require the Hardhat Runtime Environment explicitly here. This is optional
+// but useful for running the script in a standalone fashion through `node <script>`.
+//
+// You can also run a script with `npx hardhat run <script>`. If you do that, Hardhat
+// will compile your contracts, add the Hardhat Runtime Environment's members to the
+// global scope, and execute the script.
+const hre = require('hardhat');
```
Comment (Y0moo) [inline hardhat/scripts/make.js:7]: Maybe let's use the 'hardhat' name?

File: hardhat/hardhat.config.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const { url } = require("inspector");
```
Comment (Y0moo) [inline hardhat/hardhat.config.js:1]: Unused import

File: hardhat/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "hardhat-project",
+  "devDependencies": {
+    "@nomicfoundation/hardhat-toolbox": "^2.0.0",
+    "hardhat": "^2.11.2"
+  },
+  "scripts": {
+    "compile": "npx hardhat compile && rm -rf ../src/contracts && mv ./artifacts/contracts/ ../src/",
+    "chain": "npx hardhat node --network hardhat",
+    "fork": "npx hardhat node --network hardhat --fork https://mainnet.infura.io/v3/460f40a260564ac4a4f4b3fffb032dad",
```
Comment (Y0moo) [inline hardhat/package.json:10]: Please remove your infura url here

File: pages/contracts/base.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { BaseContracts, IBaseContracts } from 'components/templates/contracts/base';
+import { Default } from 'components/layouts/Default';
+import { NextPage } from 'next';
+
+const BASE: NextPage<IBaseContracts> = (props) => {
+  return (
+    <>
```
Comment (Y0moo) [inline pages/contracts/base.tsx:7]: We don't need `<></>` here

File: pages/contracts/baseMethods.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { Card } from '@web3uikit/core';
+import { FormControl, FormLabel, Input, Button, Heading, useToast } from '@chakra-ui/react';
+import { Formik } from 'formik';
+import axios from 'axios';
+import { chainId } from 'wagmi';
+
+const BaseMethods = ({ address, displayedContractFunctions, responses }) => {
```
Comment (Y0moo) [inline pages/contracts/baseMethods.tsx:8]: Missing types here

File: pages/contracts/baseResolver.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { useDropzone } from 'react-dropzone';
+import * as Lock from '../../src/contracts/Lock.sol/Lock.json';
+
+export default function BaseResolver({ contract, setContract }) {
```
Comment (Y0moo) [inline pages/contracts/baseResolver.tsx:5]: Missing types here

File: pages/contracts/baseResolver.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  useEffect(() => {
```
Comment (Y0moo) [inline pages/contracts/baseResolver.tsx:6]: I'd move this logic to separate utility to utils.ts. So we'll not have any aclculations inside .tsx file

File: src/components/elements/navigation/NavBar/paths.ts [hunk 8]
```diff
@@ -18,12 +18,24 @@ const NAV_LINKS: ISubNav[] = [
...
+        subLabel: 'Get your ERC721 and ERC1155 transfers',
         href: '/transfers/nft',
         logo: 'lazyNft',
       },
     ],
   },
+  {
+    label: 'Contracts',
+    href: '/contract',
+    children: [
```
Comment (Y0moo) [inline src/components/elements/navigation/NavBar/paths.ts:30]: Why are we using Tab with children here when it has only one subpath?

File: src/components/elements/Address.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    strokeLinecap="round"
+    strokeLinejoin="round"
+  >
+    <path stroke="none" d="M0 0h24v24H0z" fill="none" />
+    <path d="M5 12l5 5l10 -10" />
+    <title id="copied-address">Copied!</title>
+  </svg>
+);
+
+const Address = (props) => {
```
Comment (Y0moo) [inline src/components/elements/Address.tsx:31]: Missing TS here

File: src/components/elements/Address.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  useEffect(() => {
+    setAddress(props?.address);
+  }, [props]);
+
+  if (!address) {
+    return <div>Not found</div>;
+  }
+
+  const Copy = () => (
+    <svg
```
Comment (Y0moo) [inline src/components/elements/Address.tsx:44]: Please let's not use svgs like this in components, please move it to separate file

**Review by Y0moo — COMMENTED** [review]
I decided to stop reviewing since there too much things to edit. 
Missing types, ignoring TS and eslint rules, 'require' improts, package-lock.json when we have yarn.lock.
I appreciete your contribution, you need to improve code quality and I'll be happy to make a review for your code next time

**Review by Y0moo — CHANGES_REQUESTED** [review]
Please see my comments above

**Comment by BillyG83** [comment]
@3scava1i3r lets discuss this on Wednesday in our 1on1 rather than hold back @Y0moo as he has a deadline he must reach. Thanks for the feedback Dimi

**Comment by 3scava1i3r** [comment]
Understood working on the events part right now.this seems about right for the package-lock.json as i thought about it too

**Pull request closed without merging**